Dockerfile

```diff
@@ -29,12 +29,12 @@ RUN \
   touch /var/log/fpm-php.www.log && \
   chown www-data:www-data /var/log/fpm-php.www.log
 
-# Install composer 1.1.1
-RUN \
-  php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && \
-  php -r "if (hash_file('SHA384', 'composer-setup.php') === '92102166af5abdb03f49ce52a40591073a7b859a86e8ff13338cf7db58a19f7844fbc0bb79b2773bf30791e935dbd938') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;" && \
-  php composer-setup.php --version=1.1.1 --filename=composer --install-dir=/usr/local/bin && \
-  php -r "unlink('composer-setup.php');"
+# Install composer 1.1.2  using the latest installer.
+RUN curl -o /tmp/composer-installer https://getcomposer.org/installer && \
+  curl -o /tmp/composer-installer.sig https://composer.github.io/installer.sig &&  \
```

Comment:
There seem to be some confusion here.

https://composer.github.io/installer.sig is not a signature, it's a checksum. It's trivial to replace. But it's useful for checking that the file is what you expect.

The installer on the other hand, contains certificates and public keys. The former is used for the HTTPS connection to download composer (guarding aganst MITM attacs), and it then downloads the phar and a signature file and checks the signature against the phar. That's some security there.

But that doesn't help if an attacker can replace the installer, hence the checksum. A proper signature would be better, but it's probably the issue of an easy way to verify it. If it requires a working install of GPG it's unlikely to be checked by most, so i guess they opt for the easier checksum. But dynamically fetching the checksum mostly kills any security of that.